code/game/data_huds.dm

```diff
@@ -625,6 +625,68 @@
 		if(emitted_auras.Find(aura_type))
 			holder.overlays += image('icons/mob/hud.dmi', src, "hud[aura_type]aura")
 
+/mob/living/carbon/human/proc/hud_set_armor()
+	var/image/holder = hud_list[ARMOR_HUD]
+	if(!holder)
+		return
+
+	holder.overlays.Cut()
+
+	//Don't care how much armor the dead has
+	if(stat == DEAD)
+		holder.icon_state = ""
+		return
+
+	var/obj/item/clothing/armor = wear_suit ? wear_suit : null
+	if(armor?.attachments_by_slot[ATTACHMENT_SLOT_MODULE])
+		var/obj/item/armor_module/module/module = armor.attachments_by_slot[ATTACHMENT_SLOT_MODULE]
+
+		/*
+		I COULD have it do overlays += module.icon_state but if the icon_state is renamed, it would silently break
+		This way if someone renames a module's sprite, it will still work
+		The hud_icon var just has to be set whenever a new module is added
+		*/
+		if(istype(module, /obj/item/armor_module/module/chemsystem))
+			//chem_booster/proc/hud_update_resource() handles what sprite to assign
+			SEND_SIGNAL(src, COMSIG_CHEMSYSTEM_RESOURCE_UPDATE)
+		else
+			holder.overlays += image(holder.icon, src, module.hud_icon)
```

Comment:
I dunno about havinf this a signal this seems excessively convoluted (signal >component >human proc)

Rest lgtm

Reply:
The fault of the chem stuff being a component. I already hammered my head for a few hours trying to think of an alternative way but with how components are encapsulated...